lib/utilities/serverAccessLogger.js

```diff
@@ -0,0 +1,395 @@
+const { config } = require('../Config');
+const fs = require('fs');
+const os = require('os');
+const logger = require('./logger');
+
+const SERVER_ACCESS_LOG_FORMAT_VERSION = '0';
+const SERVER_ACCESS_LOGGER_HIGH_WATER_MARK = 10 * 1024 * 1024; // 10MB
+const SERVER_ACCESS_LOGGER_RETRY_REOPEN_DELAY_MS = 1000;
+const SERVER_ACCESS_LOGGER_CHECK_FILE_ROTATE_INTERVAL_MS = 10_000;
+
+var serverAccessLogger = undefined;
+var hostname = os.hostname();
+
+class ServerAccessLogger {
+    constructor(filename) {
+        this.filename = filename;
+        this.stat = undefined;
+        this._waitingDrain = false;
+        this._reopenStream();
+
+        setInterval(() => { this._checkFileRotated(); }, SERVER_ACCESS_LOGGER_CHECK_FILE_ROTATE_INTERVAL_MS);
```

Comment:
`setInterval` is not cleared, it might not be an issue since we create one ServerAccessLogger instance per process, right?